flink-core/src/main/java/org/apache/flink/api/common/typeutils/base/BigIntSerializer.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.common.typeutils.base;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.core.memory.DataInputView;
+import org.apache.flink.core.memory.DataOutputView;
+
+@Internal
+public final class BigIntSerializer extends TypeSerializerSingleton<BigInteger> {
+
+	private static final long serialVersionUID = 1L;
+
+	public static final BigIntSerializer INSTANCE = new BigIntSerializer();
+
+	@Override
+	public boolean isImmutableType() {
+		return true;
+	}
+
+	@Override
+	public BigInteger createInstance() {
+		return BigInteger.ZERO;
+	}
+
+	@Override
+	public BigInteger copy(BigInteger from) {
+		return from;
+	}
+	
+	@Override
+	public BigInteger copy(BigInteger from, BigInteger reuse) {
+		return from;
+	}
+
+	@Override
+	public int getLength() {
+		return -1;
+	}
+
+	@Override
+	public void serialize(BigInteger record, DataOutputView target) throws IOException {
+		writeBigInteger(record, target);
+	}
+
+	@Override
+	public BigInteger deserialize(DataInputView source) throws IOException {
+		return readBigInteger(source);
+	}
+	
+	@Override
+	public BigInteger deserialize(BigInteger reuse, DataInputView source) throws IOException {
+		return readBigInteger(source);
+	}
+
+	@Override
+	public void copy(DataInputView source, DataOutputView target) throws IOException {
+		copyBigInteger(source, target);
+	}
+
+	@Override
+	public boolean canEqual(Object obj) {
+		return obj instanceof BigIntSerializer;
+	}
+
+	// --------------------------------------------------------------------------------------------
+	//                           Static Helpers for BigInteger Serialization
+	// --------------------------------------------------------------------------------------------
+
+	public static void writeBigInteger(BigInteger record, DataOutputView target) throws IOException {
+		// null value support
+		if (record == null) {
+			target.writeInt(0);
+			return;
+		}
+		// fast paths for 0, 1, 10
+		else if (record == BigInteger.ZERO) {
```

Comment:
Checks with equals would be too expensive. This should only be a reference check. Kryo does the same.